File: Two Pointers/3194. Minimum Average of Smallest and Largest Elements/3194. Minimum Average of Smallest and Largest Elements.py
class Solution:
    def minimumAverage(self, nums: List[int]) -> float:
        nums.sort()
        min_avg = float("inf")
        l = 0
        r = len(nums) - 1

        while l < r:
            min_avg = min(min_avg, (nums[l] + nums[r]) / 2)
            l += 1
            r -= 1
        
        return min_avg

# Time: O(n log n)
# Space: O(1)

File: Two Pointers/3194. Minimum Average of Smallest and Largest Elements/3194. Minimum Average of Smallest and Largest Elements.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var minimumAverage = function (nums) {
  nums.sort((a, b) => a - b);
  let min_average = Infinity;
  let l = 0;
  let r = nums.length - 1;

  while (l < r) {
    min_average = Math.min(min_average, (nums[l] + nums[r]) / 2);
    l++;
    r--;
  }

  return min_average;
};

// Time: O(n log n)
// Space: O(1)
